stringutil/stringutil.go

```diff
@@ -278,3 +278,20 @@ func ObfuscateLeft(raw string, number int, obfuscator string) string {
 func ObfuscateRight(raw string, number int, obfuscator string) string {
 	return Obfuscate(raw, number, Right, obfuscator)
 }
+
+// ObfuscateRightPercent obfuscates percent of string, leaves at least 1 character
+func ObfuscateRightPercent(raw string, percent int, obfuscator string) string {
+
+	percent = intutil.Range(percent, 0, 100)
+
+	l := len(raw) - intutil.Max(1, len(raw)-len(raw)*percent/100)
+	return ObfuscateRight(raw, l, obfuscator)
+}
+
+// ObfuscateLeftPercent obfuscates percent of string, leaves at least 1 character
+func ObfuscateLeftPercent(raw string, percent int, obfuscator string) string {
+	percent = intutil.Range(percent, 0, 100)
+
+	l := len(raw) - intutil.Max(1, len(raw)-len(raw)*percent/100)
+	return ObfuscateLeft(raw, l, obfuscator)
+}
```

Comment:
DRY opportunity with percentage calculation:
```suggestion
// ObfuscateRightPercent obfuscates percent of string, leaves at least 1 character
func ObfuscateRightPercent(raw string, percent int, obfuscator string) string {
	return ObfuscatePercent(raw, percent, Right, obfuscator)
}

// ObfuscateLeftPercent obfuscates percent of string, leaves at least 1 character
func ObfuscateLeftPercent(raw string, percent int, obfuscator string) string {
	return ObfuscatePercent(raw, percent, Left, obfuscator)
}

func ObfuscatePercent(raw string, percent int, direction Direction, obfuscator string) string {
	percent = intutil.Range(percent, 0, 100)
	l := len(raw) - intutil.Max(1, len(raw)-len(raw)*percent/100)
	return Obfuscate(raw, l, direction, obfuscator)
}
```